maintenance/wikia/updateDatawarePages.php

```diff
@@ -0,0 +1,290 @@
+			[
+				'page_id',
+				'page_is_redirect',
+				'page_latest',
+				'page_title',
+				'page_namespace',
+				'count(rev_id) as page_edits',
+				'max(rev_timestamp) as page_last_edited',
+				'rev_user'
+			],
+			[ ],
+			__METHOD__,
+			[
+				'GROUP BY' => 'page_id',
+				'ORDER BY' => 'page_id',
+			],
+			[
+				'revision' => [ 'INNER JOIN', 'rev_page = page_id' ],
+			]
+		);
+
+		$pages = array();
+		/** @var stdClass $row */
+		foreach ( $res as $row ) {
+			$row->page_latest = intval( $row->page_latest );
+			$row->page_edits = intval( $row->page_edits );
+			$row->page_is_redirect = intval( $row->page_is_redirect );
+			$row->page_is_content = intval( in_array( $row->page_namespace, $wgContentNamespaces ) );
+			$row->page_last_edited = $row->page_last_edited ? wfTimestamp(TS_DB, $row->page_last_edited) : null;
+			$this->attachTextualTitle($row);
+			$pages[$row->page_id] = $row;
+		}
+		$res->free();
+
+		$this->output("Got " . count($pages) . " local pages.\n");
+
+		return $pages;
+	}
+
+	private function attachTextualTitle( $page ) {
+		$nsText = '';
+		if ( $page->page_namespace ) {
+			$nsText = MWNamespace::getCanonicalName( $page->page_namespace );
+			if ($nsText) {
+				$nsText = $page->page_namespace;
+			}
+			$nsText .= ':';
+		}
+
+		$page->page_textual_title = "{$nsText}{$page->page_title} (id={$page->page_id})";
+	}
+
+	private function getDatawarePages() {
+		global $wgCityId, $wgExternalDatawareDB;
+
+		$this->output("Fetching page index from dataware...\n");
+
+		$dbr = wfGetDB( DB_SLAVE, null, $wgExternalDatawareDB );
+		$res = $dbr->select(
+			'pages',
+			[
+				'page_id',
+				'page_status',
+				'page_latest',
+				'page_title',
+				'page_title_lower',
+				'page_namespace',
+				'page_is_redirect',
+				'page_is_content',
+				'page_edits',
+				'page_last_edited'
+			],
+			[
+				'page_wikia_id' => $wgCityId,
+			],
+			__METHOD__
+		);
+
+		$pages = [ ];
+		/** @var stdClass $row */
+		foreach ( $res as $row ) {
+			$this->attachTextualTitle($row);
+			$pages[$row->page_id] = $row;
+		}
+		$res->free();
+
+		$this->output("Got " . count($pages) . " pages from dataware.\n");
+
+		return $pages;
+	}
+
+	private function comparePages( $localPages, $datawarePages ) {
+		global $wgExternalDatawareDB;
+
+		$ids = array_merge( array_keys( $localPages ), array_keys( $datawarePages ) );
+		$ids = array_unique( $ids );
+
+		$this->output("Comparing pages between local list and index...\n");
+		$dbw = wfGetDB( DB_MASTER, null, $wgExternalDatawareDB );
+		$stats = [
+			'added' => 0,
+			'updated' => 0,
+			'removed' => 0,
+		];
+		foreach ( $ids as $id ) {
+			$localPage = isset( $localPages[$id] ) ? $localPages[$id] : null;
+			$datawarePage = isset( $datawarePages[$id] ) ? $datawarePages[$id] : null;
+
+			$this->comparePage( $stats, $dbw, $localPage, $datawarePage );
+		}
+		$this->output("Update statistics: added={$stats['added']} updated={$stats['updated']} removed={$stats['removed']}.\n");
```

Comment:
Let's push stats as a part of a context here as well. This will allow us to check script executions that made big changes to the `pages` table - this can indicate failing on edit triggered updates made by Perl backend.